src/js/category.js

```diff
@@ -0,0 +1,256 @@
+const listProduct = document.querySelector('.category__product-list');
+const gridProduct = document.querySelector('.category__product-grid');
+const menuCategory = document.querySelector('.filter__nav');
+let categoryApplied = '';
+let menu = [];
+let products = [];
+
+class Api {
+  getProduct = async function (page = 1, limit = 9) {
+    try {
+      //đợi lệnh gọi fetch() được giải quyết xong
+      //Lệnh gọi fetch() sẽ trả về một phản hồi hoặc tạo ra một lỗi
+      const res = await fetch(
+        `http://localhost:3000/product?_page=${page}&_limit=${limit}`
+      );
+      //đợi lệnh gọi res.json() được giải quyết
+      //trả về đối tượng JSON hoặc gây ra lỗi
+      const data = await res.json();
+      if (!res.ok) throw new Error(`HTTP error: ${res.status}`);
```

Comment:
Thông thường thì xuất hiện lỗi khi call API thì hầu như là bị từ bên phía server
Ở đây em đang bị lặp code ở đoạn check status res. Em có tách nó ra 1 hàm riêng
VD: 
```
function handleErrors(response) {
    if (!response.ok) {
        throw Error(`HTTP error: ${response.statusText}`);
    }
    return response;
}
```
Lúc này sẽ là
```suggestion
      const res = await fetch(
        `http://localhost:3000/product?_page=${page}&_limit=${limit}`
      ).then(handleErrors);
      const data = await res.json();
```